packages/component-library/src/components/plmg-button/plmg-button.scss

```diff
@@ -0,0 +1,394 @@
+        tokens.$plmg-background-danger-strong-hover,
+        tokens.$plmg-background-danger-strong-hover,
+        tokens.$plmg-text-neutral-inverted
+      );
+    }
+    &:active {
+      @include background-border-text(
+        tokens.$plmg-background-danger-strong-active,
+        tokens.$plmg-background-danger-strong-active,
+        tokens.$plmg-text-neutral-inverted
+      );
+    }
+    &:focus {
+      @include background-border-text(
+        tokens.$plmg-background-danger-strong-hover,
+        tokens.$plmg-border-standout,
+        tokens.$plmg-text-neutral-inverted,
+        tokens.$plmg-border-width-l
+      );
+    }
+  }
+}
+
+@mixin outline-color-background {
+  &.primary {
+    @include border-background-none(tokens.$plmg-border-primary-strong);
+
+    &:hover {
+      @include border-background-none(tokens.$plmg-border-primary);
+    }
+    &:active {
+      @include background-border(
+        tokens.$plmg-background-primary,
+        tokens.$plmg-border-primary
+      );
+    }
+  }
+  &.neutral {
+    @include border-background-none(tokens.$plmg-border-neutral);
+
+    &:hover {
+      @include border-background-none(tokens.$plmg-border-primary);
+    }
+    &:active {
+      @include background-border(
+        tokens.$plmg-background-neutral-active,
+        tokens.$plmg-border-neutral
+      );
+    }
+  }
+  &.standout {
+    @include border-background-none(tokens.$plmg-border-standout-strong);
+
+    &:hover {
+      @include border-background-none(tokens.$plmg-border-standout);
+    }
+    &:active {
+      @include background-border(
+        tokens.$plmg-background-standout,
+        tokens.$plmg-border-standout
+      );
+    }
+  }
+  &.danger {
+    @include border-background-none(tokens.$plmg-border-danger-strong);
+
+    &:hover {
+      @include border-background-none(tokens.$plmg-border-danger);
+    }
+    &:active {
+      @include background-border(
+        tokens.$plmg-background-danger,
+        tokens.$plmg-border-danger
+      );
+    }
+  }
+
+  &:focus {
+    @include border-background-none(
+      tokens.$plmg-border-standout,
+      tokens.$plmg-border-width-l
+    );
+  }
+}
+
+@mixin outline-color-text {
+  &.primary {
+    @include just-text(tokens.$plmg-text-primary-strong);
+
+    &:hover {
+      @include just-text(tokens.$plmg-text-primary-strong-hover);
+    }
+    &:active {
+      @include just-text(tokens.$plmg-text-primary-strong-active);
+    }
+    &:focus {
+      @include just-text(tokens.$plmg-text-primary-strong-hover);
+    }
+  }
+  &.neutral {
+    @include just-text(tokens.$plmg-text-neutral);
+  }
+  &.standout {
+    @include just-text(tokens.$plmg-text-standout);
+  }
+  &.danger {
+    @include just-text(tokens.$plmg-text-danger-strong);
+
+    &:active {
+      @include just-text(tokens.$plmg-text-danger-strong-active);
+    }
+  }
+}
+// END Mixins
+// ------------------------------
+
+// Component style
+// ------------------------------
+button {
```

Comment:
I would prefer to use a class name here, such as `.button`. This way, we only apply styles where explicitly stated. Also, if we were to extract this styling to a general-purpose library (say for Wordpress or asp.net) we could simply include the CSS and do `class="button primary medium"` and it should all work perfectly fine even outside of web components. Just a thought, as I don't know if there is a reason if this must be a HTML element for some magic to work...

Reply:
With Stencil and web components, each component will live in its own shadow-dom. Therefore, all styles will be isolated from each other. That's why we can be very generic and use `button`.